tests/multiConnection.ts

```diff
@@ -0,0 +1,309 @@
+	private enabled = true;
+	public callCount = 0;
+
+	constructor(connection: Connection) {
+		super(connection.rpcEndpoint);
+		this.connection = connection;
+	}
+
+	async getMultipleAccountsInfoAndContext(
+		publicKeys: PublicKey[],
+		commitment?: Commitment
+	): Promise<RpcResponseAndContext<(AccountInfo<Buffer> | null)[]>> {
+		if (this.enabled) {
+			this.callCount++;
+			return this.connection.getMultipleAccountsInfoAndContext(
+				publicKeys,
+				commitment
+			);
+		} else {
+			return Promise.resolve(null);
+		}
+	}
+
+	enable() {
+		this.enabled = true;
+	}
+
+	disable() {
+		this.enabled = false;
+	}
+}
+
+describe('multiConnection', () => {
+	const provider = anchor.AnchorProvider.local();
+	const connection = provider.connection;
+	anchor.setProvider(provider);
+	const chProgram = anchor.workspace.ClearingHouse as Program;
+
+	let adminClearingHouse: Admin;
+	const eventSubscriber = new EventSubscriber(connection, chProgram);
+	eventSubscriber.subscribe();
+
+	let usdcMint;
+	const usdcAmount = new BN(1000000 * 10 ** 6);
+
+	// ammInvariant == k == x * y
+	const mantissaSqrtScale = new BN(Math.sqrt(MARK_PRICE_PRECISION.toNumber()));
+	const ammInitialQuoteAssetReserve = new anchor.BN(5 * 10 ** 13).mul(
+		mantissaSqrtScale
+	);
+	const ammInitialBaseAssetReserve = new anchor.BN(5 * 10 ** 13).mul(
+		mantissaSqrtScale
+	);
+
+	before(async () => {
+		usdcMint = await mockUSDCMint(provider);
+
+		const solUsd = await mockOracle(1);
+		const periodicity = new BN(60 * 60); // 1 HOUR
+
+		adminClearingHouse = new Admin({
+			connection,
+			wallet: provider.wallet,
+			programID: chProgram.programId,
+			opts: {
+				commitment: 'confirmed',
+			},
+			activeUserId: 0,
+			marketIndexes: [new BN(0)],
+			bankIndexes: [new BN(0)],
+			oracleInfos: [{ publicKey: solUsd, source: OracleSource.PYTH }],
+		});
+		await adminClearingHouse.initialize(usdcMint.publicKey, true);
+		await adminClearingHouse.subscribe();
+
+		await initializeQuoteAssetBank(adminClearingHouse, usdcMint.publicKey);
+		await adminClearingHouse.updateAuctionDuration(new BN(0), new BN(0));
+
+		await adminClearingHouse.initializeMarket(
+			solUsd,
+			ammInitialBaseAssetReserve,
+			ammInitialQuoteAssetReserve,
+			periodicity
+		);
+
+		await adminClearingHouse.fetchAccounts();
+	});
+
+	after(async () => {
+		await adminClearingHouse.unsubscribe();
+		await eventSubscriber.unsubscribe();
+	});
+
+	it('Can still update user accounts', async () => {
+		const firstConnection = new mockConnection(provider.connection);
+		const secondConnection = new Connection('http://second-connection.com');
+		const myMultiConn = new MultiConnection([
+			firstConnection,
+			secondConnection,
+		]);
+		const bulkAccountLoader = new BulkAccountLoader(
+			myMultiConn,
+			'confirmed',
+			1000
+		);
+		// const config = initialize({ env: 'devnet' });
+
+		const userKeypair = new Keypair();
+
+		const clearingHouse = new ClearingHouse({
+			connection: firstConnection,
```

Comment:
does there remain a potential problem where the connection the clearing house directly uses could still fail and not recover? Don't know if we've ever had that issue before or not .. but if I understand correctly, this PR fixes the "fetching" side of things with the account loaders but we could still get stuck behind a broken rpc for other stuff in the clearinghouse?

Reply:
yeah exactly, motivation for this feature is to keep the account loader up and running. Not really sure how to best handle a completely down RPC, maybe something like https://github.com/drift-labs/protocol-v2/pull/66#discussion_r956491485 is the step in the right direction to prevent trying to use a down'd RPC